scripts/microbe_stage/organelle.as

```diff
@@ -107,6 +107,9 @@ class Organelle{
         Hex@ hex = Hex(q, r);
 
         @hexes[formatInt(s)] = hex;
+
+        //update hex cache
+        cachedHexCount += 1;
```

Comment:
It's not always guaranteed that the line 109 inserts a new hex (it could overwrite an existing one).

Reply:
Shouldn't it then return false earlier?

Reply:
So it does. I was just quickly looking at the diff shown on github.

Reply:
Is there anything else that needs to be fixed for this issue?

Reply:
I don't see anywhere the player cell's cache being reset. Also I think the cache should be reset when an organelle is added or removed from a cell (when it is preparing to split).

Reply:
Can the hex count be changed on the organelle?
I only see an add function and if I understand it correctly the hex collection isn't exposed but array's of it are constructed. Not saying you're not right just that I'll have to understand the code better to know where I can change what you're saying.

Reply:
The hex count of individual organelle type is never changed. But cells duplicate their organelles when they get ready to split and again after splitting they reset the cell layout. Additionally when the player makes changes in the editor that has to update the size of their cell (the player cell entity is never recreated)

Reply:
So would an update need to happen in placed organelle or in microbe?

Reply:
The organelle adding and removing is in MicrobeOperations namespace (microbe_operations.as)

Reply:
Should be done ion my new commit